src/main/java/eu/interop/federationgateway/service/DiagnosisKeyBatchService.java

```diff
@@ -65,158 +61,140 @@
 
-  private boolean checkBatchSize(List<DiagnosisKeyEntity> diagnosisKeyCollector) {
-    if (diagnosisKeyCollector.size() > properties.getBatching().getDoclimit()) {
-      log.info("Continue the batching process to next batch while {} "
-          + "keys collected and the limit: {} of keys is reached",
-        diagnosisKeyCollector.size(), properties.getBatching().getDoclimit());
-      return true;
-    }
-    return false;
+
+    MDC.remove("batchCount");
+    MDC.remove("batchTime");
   }
 
-  private boolean checkUploaderLimitation(List<DiagnosisKeyEntity> diagnosisKeyEntitys, String uploaderBatchTag) {
-    if (diagnosisKeyEntitys.size() > properties.getBatching().getDoclimit()) {
-      log.error("Stop batching process, while try to batch {} keys from one uploader: {}, "
-          + "but the configured batching limit of keys is {}",
-        diagnosisKeyEntitys.size(), uploaderBatchTag, properties.getBatching().getDoclimit());
-      return true;
+  /**
+   * Creates a new Batch Entity and sets batchTag to all contained diagnosiskeys.
+   *
+   * @return true if a batch was created or false if not.
+   */
+  @Transactional(Transactional.TxType.REQUIRES_NEW)
+  public boolean createNextBatch() {
+
+    List<String> uploaderBatchTags = collectUploaderBatchTags();
+    log.info("collected batch tags");
+
+    if (uploaderBatchTags.isEmpty()) {
+      log.info("successfully finish the document batching process - no more unprocessed diagnosis keys left");
+      return false;
     }
-    return false;
+
+    DiagnosisKeyBatchEntity newBatchEntity = createNextBatchEntityAndLinkPredecessor();
+
+    int updatedRows = diagnosisKeyEntityRepository.setBatchTagByUploaderBatchTag(
+      uploaderBatchTags, newBatchEntity.getBatchName());
+
+    callbackService.notifyAllCountriesForNewBatchTag(newBatchEntity);
+
+
+    MDC.put("batchTag", newBatchEntity.getBatchName());
+    MDC.put("diagnosisKeyCount", String.valueOf(updatedRows));
+    log.info("Batch created");
+    MDC.remove("diagnosisKeyCount");
+    MDC.remove("batchTag");
+
+    return true;
   }
 
-  private boolean checkFormat(List<DiagnosisKeyEntity> diagnosisKeyEntitys, Optional<DiagnosisKeyEntity> diagnosisKey) {
+  /**
+   * Creates a new instance of DiagnosisKeyBatchEntity und generates a batchTag for it.
+   * Also the preceding batch entity will be linked to the newly created batch entity.
+   *
+   * @return the created DiagnosisKeyBatchEntity.
+   */
+  @Transactional(Transactional.TxType.MANDATORY)
+  public DiagnosisKeyBatchEntity createNextBatchEntityAndLinkPredecessor() {
+    DiagnosisKeyBatchEntity newBatchEntity = new DiagnosisKeyBatchEntity();
+    newBatchEntity.setCreatedAt(ZonedDateTime.now(ZoneOffset.UTC));
+
+    String formattedDate = LocalDate.now().format(DateTimeFormatter.ofPattern("yyyyMMdd"));
+
+    Optional<DiagnosisKeyBatchEntity> lastBatchEntity = diagnosisKeyBatchRepository.findTopByOrderByCreatedAtDesc();
 
-    if (diagnosisKeyEntitys.stream()
```

Comment:
do not forget the requirement: make sure that within a batch the versions (FormatInformation, majorVersion) of the uploaded data are the same
proposal: separate new versions in separate batches